src/services/transaction_reversal.rs

```diff
@@ -1,154 +1,138 @@
+use derive_builder::Builder;
 use serde::{Deserialize, Serialize};
+use url::Url;
 
 use crate::{ApiEnvironment, CommandId, IdentifierTypes, Mpesa, MpesaError, MpesaResult};
 
 const TRANSACTION_REVERSAL_URL: &str = "mpesa/reversal/v1/request";
 
 #[derive(Debug, Serialize)]
-pub struct TransactionReversalPayload<'mpesa> {
-    #[serde(rename(serialize = "Initiator"))]
-    initiator: &'mpesa str,
-    #[serde(rename(serialize = "SecurityCredential"))]
-    security_credentials: &'mpesa str,
-    #[serde(rename(serialize = "CommandID"))]
-    command_id: CommandId,
-    #[serde(rename(serialize = "TransactionID"))]
-    transaction_id: &'mpesa str,
-    #[serde(rename(serialize = "ReceiverParty"))]
-    receiver_party: &'mpesa str,
-    #[serde(rename(serialize = "RecieverIdentifierType"))]
-    receiver_identifier_type: IdentifierTypes,
-    #[serde(rename(serialize = "ResultURL"))]
-    result_url: &'mpesa str,
-    #[serde(rename(serialize = "QueueTimeOutURL"))]
-    timeout_url: &'mpesa str,
-    #[serde(rename(serialize = "Remarks"))]
-    remarks: &'mpesa str,
-    #[serde(rename(serialize = "Occasion"))]
-    occasion: &'mpesa str,
-    #[serde(rename(serialize = "Amount"))]
-    amount: f64,
+#[serde(rename_all = "PascalCase")]
+pub struct TransactionReversalRequest<'mpesa> {
+    /// The name of the initiator to initiate the request.
+    pub initiator: &'mpesa str,
+    /// Encrypted Credential of user getting transaction reversed.
+    pub security_credential: String,
+    /// Unique command for each transaction type.
+    #[serde(rename = "CommandID")]
+    pub command_id: CommandId,
+    /// This is the Mpesa Transaction ID of the transaction which you wish to
+    #[serde(rename = "TransactionID")]
+    pub transaction_id: &'mpesa str,
+    /// The organization that receives the transaction.
+    pub receiver_party: &'mpesa str,
+    /// Type of organization that receives the transaction.
+    pub receiver_identifier_type: IdentifierTypes,
+    /// The path that stores information about the transaction.
+    #[serde(rename = "ResultURL")]
+    pub result_url: Url,
+    /// The path that stores information about the time-out transaction.
+    #[serde(rename = "QueueTimeOutURL")]
+    pub queue_timeout_url: Url,
+    /// Comments that are sent along with the transaction.
+    pub remarks: &'mpesa str,
+    /// Comments that are sent along with the transaction.
+    pub occasion: Option<&'mpesa str>,
+    /// The amount transacted in the transaction is to be reversed, down to the
+    /// cent.
+    pub amount: f64,
```

Comment:
Decimal numbers are considered invalid by Safaricom, we should be using whole numbers here

Reply:
What bothers is from the documentation they wrote down to the cent